### ISSUE TITLE
📝 Error messages for `selfCloseVoidElements` contradicts itself.

### ISSUE THREAD

**Issue by lgarron** [biomejs/biome#7882]
The error for [`html.formatter.selfCloseVoidElements `](https://biomejs.dev/reference/configuration/#htmlformatterselfclosevoidelements) includes the following:

- ✖ Expected a matching closing tag but instead found '</style>'.
- ℹ Expected a matching closing tag here.
- ✖ Void elements should not have a closing tag.

These messages straight-up contradict themselves.

Presumably https://github.com/biomejs/biome/blob/be79a6b41bf365535296d88fca3e9a235817458f/crates/biome_html_parser/src/syntax/parse_error.rs#L63 is being triggered incorrectly.

### Environment information

```bash
CLI:
  Version:                      2.3.1
  Color support:                true

Platform:
  CPU Architecture:             aarch64
  OS:                           macos

Environment:
  BIOME_LOG_PATH:               unset
  BIOME_LOG_PREFIX_NAME:        unset
  BIOME_CONFIG_PATH:            unset
  BIOME_THREADS:                unset
  NO_COLOR:                     unset
  TERM:                         xterm-256color
  JS_RUNTIME_VERSION:           v24.10.0
  JS_RUNTIME_NAME:              node
  NODE_PACKAGE_MANAGER:         bun/1.3.1

Biome Configuration:
  Status:                       Loaded successfully
  Path:                         biome.json
  Formatter enabled:            true
  Linter enabled:               true
  Assist enabled:               true
  VCS enabled:                  true

Formatter:
  Format with errors:           unset
  Indent style:                 Space
  Indent width:                 2
  Line ending:                  unset
  Line width:                   unset
  Attribute position:           unset
  Bracket spacing:              unset
  Includes:                     unset

JavaScript Formatter:
  Enabled:                      unset
  JSX quote style:              unset
  Quote properties:             unset
  Trailing commas:              unset
  Semicolons:                   unset
  Arrow parentheses:            unset
  Bracket spacing:              unset
  Bracket same line:            unset
  Quote style:                  unset
  Indent style:                 unset
  Indent width:                 unset
  Line ending:                  unset
  Line width:                   unset
  Attribute position:           unset

JSON Formatter:
  Enabled:                      unset
  Indent style:                 unset
  Indent width:                 unset
  Line ending:                  unset
  Line width:                   unset
  Trailing Commas:              unset
  Expand lists:                 unset

CSS Formatter:
  Enabled:                      unset
  Indent style:                 unset
  Indent width:                 unset
  Line ending:                  unset
  Line width:                   unset
  Quote style:                  unset

GraphQL Formatter:
  Enabled:                      unset
  Indent style:                 unset
  Indent width:                 unset
  Line ending:                  unset
  Line width:                   unset
  Bracket spacing:              unset
  Quote style:                  unset

Workspace:
  Open Documents:               0
```

### Configuration

N/A

### Playground link

https://biomejs.dev/playground/?analyzerFixMode=safeAndUnsafeFixes&code=PABmAGkAZQBsAGQAcwBlAHQAPgAKACAAIAA8AHMAdQBtAG0AYQByAHkAPgBBAGwAZwAgADwAYgB1AHQAdABvAG4AIABpAGQAPQAiAHMAZQB0AC0AcwBlAGEAcgBjAGgALQBhAGwAZwAiAD4AQQBwAHAAbAB5ACAAYQBsAGcAIAB0AG8AIABkAGUAZgBhAHUAbAB0ACAAcABhAHQAdABlAHIAbgAgAGEAbgBkACAAcwBlAHQAIABhAHMAIABzAGUAYQByAGMAaAAgAGcAbwBhAGwAPAAvAGIAdQB0AHQAbwBuAD4APAAvAHMAdQBtAG0AYQByAHkAPgAKACAAIAA8AGIAcgA%2BAAoAIAAgADwAaQBuAHAAdQB0ACAAaQBkAD0AIgBzAGUAYQByAGMAaAAtAGEAbABnACIAIABwAGwAYQBjAGUAaABvAGwAZABlAHIAPQAiAEEAbABnACIAPgA8AC8AaQBuAHAAdQB0AD4ACgA8AC8AZgBpAGUAbABkAHMAZQB0AD4A&language=html&ruleDomains.react=recommended&ruleDomains.test=recommended&ruleDomains.next=recommended&ruleDomains.project=recommended

### Code of Conduct

- [x] I agree to follow Biome's Code of Conduct

**Comment by lgarron** [comment]
Oh! I think I understand what's going on.

When Biome sees `</input>`, it:

- recognizes it as a closing tag for `<input>` and shows an error for this, but 
- *also* tries to interpret it as a closing tag for the parent element.

I think it basically never makes sense to show the latter error if the former error is shown — that is, the `</input>` should be consumed by the parsing/error reporting before it gets to the second error.

**Comment by dyc3** [comment]
Yeah the diagnostics should be better here. Thanks for the report!
